### PR TITLE
Shape squeeze edge case bug fix

### PR DESCRIPTION
**Pull request by benidis**
Fixes an edge case where we pass a single value and `np.squeeze` suppresses all dimensions.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/gluonts/transform/feature.py [hunk 1]
```diff
@@ -99,6 +99,8 @@ class LastValueImputation(MissingValueImputation):
     """
 
     def __call__(self, values: np.ndarray) -> np.ndarray:
+        if len(values) == 1:
+            return DummyValueImputation()(values)
```
Comment (lostella) [inline src/gluonts/transform/feature.py:103]: This would need to be converted to a float I guess

```suggestion
            return DummyValueImputation()(values.item())
```

Comment (benidis) [reply src/gluonts/transform/feature.py:103]: No it should be numpy array: `def __call__(self, values: np.ndarray) -> np.ndarray:`. 

A simple example (copying what the dummy value imputation does) with numpy array:
```
values = np.array([1])
nan_indices = np.where(np.isnan(values))
values[nan_indices] = 0
print(values)
[1]
```

and with float:
```
values = np.array([1]).item()
nan_indices = np.where(np.isnan(values))
values[nan_indices] = 0
print(values)

---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-97-8448c7edc37d> in <module>
      1 values = np.array([1]).item()
      2 nan_indices = np.where(np.isnan(values))
----> 3 values[nan_indices] = 0
      4 print(values)

TypeError: 'int' object does not support item assignment

```

**Comment by benidis** [comment]
> Is this edge case covered by tests?

No, I am adding a test.

File: src/gluonts/transform/feature.py (hunk 1 above)
Comment (lostella) [reply src/gluonts/transform/feature.py:105]: yes, sorry, I got confused between __init__ and __call__

File: src/gluonts/transform/feature.py [hunk 2]
```diff
@@ -87,6 +87,8 @@ class MeanValueImputation(MissingValueImputation):
...
```
Comment (lostella) [inline src/gluonts/transform/feature.py:91]: Would it make sense to pass a user specified dummy_value when constructing the DummyValueImputation object here? Essentially, all these classes would allow specifying the dummy_value, to be used in extreme cases.

But maybe not urgent and can be addressed later.

Comment (benidis) [reply src/gluonts/transform/feature.py:91]: Yeah, so the `DummyValueImputation` accepts a value that uses to impute. With the current implementation it uses the default. We can always add one more argument to all the imputation methods to override the default but maybe this is too much. If needed we can always include it easily.
